chainer/functions/tensor_network.py

```diff
@@ -0,0 +1,264 @@
+import math
+
+import numpy
+
+from chainer import cuda
+from chainer import function
+from chainer.utils import type_check
+
+
+def _as_vec(x):
+    return x.reshape(x.size)
+
+
+def _as_mat(x):
+    return x.reshape(x.shape[0], x.size // x.shape[0])
+
+
+def _empty_like(x):
+    if isinstance(x, cuda.GPUArray):
+        return cuda.empty_like(x)
+    else:
+        return numpy.empty_like(x)
+
+
+class TensorNetwork(function.Function):
+
+    def __init__(self, in_shape, out_size, nobias=False,
+                 initialW=None, initial_bias=None):
+
+        self.W = None
+        self.gW = None
+        self.V1 = None
+        self.gV1 = None
+        self.V2 = None
+        self.gV2 = None
+        self.b = None
+        self.gb = None
+
+        assert len(in_shape) == 2
+        self.in_shape = in_shape
+
+        if initialW is not None:
+            assert initialW.shape == (in_shape[0], in_shape[1], out_size)
+            self.W = initialW
+        else:
+            in_size = numpy.prod(self.in_shape)
+            self.W = numpy.random.normal(
+                0, math.sqrt(1. / in_size),
+                (self.in_shape[0], self.in_shape[1], out_size)
+            ).astype(numpy.float32)
+
+        if initial_bias is not None:
+            assert len(initial_bias) == 3
+            self.V1, self.V2, self.b = initial_bias
+        elif not nobias:
+            self.V1 = numpy.zeros(
+                (self.in_shape[0], out_size), dtype=numpy.float32)
+            self.V2 = numpy.zeros(
+                (self.in_shape[1], out_size), dtype=numpy.float32)
+            self.b = numpy.zeros((out_size, ), dtype=numpy.float32)
+
+        self.gW = _empty_like(self.W)
+        if not nobias:
```

Comment:
My expectation was `nobias` option is stronger than `initial_bias`. So this code is inappropriate.